api/envoy/api/v2/route/route.proto

```diff
@@ -429,14 +429,14 @@ message CorsPolicy {
     //   :ref:`filter_enabled<envoy_api_field_route.CorsPolicy.filter_enabled>` field instead.
     google.protobuf.BoolValue enabled = 7 [deprecated = true];
 
-    // Specifies if CORS is enabled.
+    // Specifies if CORS is enabled. Default is disabled.
```

Comment:
This should be [enabled by default](https://github.com/envoyproxy/envoy/blob/02ed57d648d82c5e5ef72070b145d2a28c9b404c/source/common/router/config_impl.cc#L140) if absent from the config. The default value in the documentation for `filter_enabled` mirrors the documentation for the `enabled` value in that if either are absent, the default behavior should be that the filter is enabled.

The CORS sandbox has [an example of this](https://github.com/envoyproxy/envoy/blob/master/examples/cors/backend/front-envoy.yaml#L57-L60). The `/cors/restricted` route defines a CORS policy without the `enabled` and `filter_enabled` keys.

Reply:
Thank you for the example that neither `enabled` nor `filter_enable` exists. I didn't think about this possibility.

However, this comments referring to the case that
- `enabled` is missing, and
- `filter_enable` exists but with default value.
Something like https://github.com/envoyproxy/envoy/blob/master/examples/cors/backend/front-envoy.yaml#L48-L49 *WITHOUT* line 50-52
```
                  cors:
                    filter_enabled:
```
The above case is a legit proto value but CORS is disabled.

Reply:
`filter_enabled` is a `RuntimeFractionalPercent`, which requires that it has a [`default_value`](https://github.com/envoyproxy/envoy/blob/02ed57d648d82c5e5ef72070b145d2a28c9b404c/api/envoy/api/v2/core/base.proto#L242-L244) key/value pair. I believe the above would throw an exception.

Reply:
@dschaller Make sense... Let me double check my test case in #6570

Reply:
That test case is misleading.

This comment is also misleading... as there is no such a default value of `filter_enabled` because you have to specify an explicit `RuntimeFractionalPercent`.Otherwise the default constructed RuntimeFractionalPercent is failing at the validation which requires `default_value`.

Reply:
Not sure I agree about it being misleading. It could be that the comment is causing confusion because there are two possible fields to enable this filter. The `filter_enabled` key is equivalent to a default value of `100/HUNDRED` if the key is absent. This is true since if that key is not there and enabled is not present, the CORS policy will use the legacy value of `enabled`, which defaults to `true`.

The `default_value` is not documented here as required since it belongs to `RuntimeFractionalPercent`, which documents that the value is enabled and does not have a default value.

Reply:
@dschaller I think that is problematic since it does not conform to the standard proto policy of unset message being the default message, and an unset int field being 0. So the correct default would be 0.

Your explanation supports the fact that CORS enabled if using `filter_enabled: {}`. But I'm concerned what's going to happen once the legacy `enabled` and code branches are removed, and the default changes from 100 to 0.

Reply:
@kyessenov I see your point here but that is more of an implementation issue. The current behavior assumes that if you have a CORS policy defined in your envoy configuration it will be enabled for routes. For what it's worth, I don't believe you would end up in scenario where the `filter_enabled`'s `default_value` key is not set, since the proto requires that key have a value. :)

The proposed changes to the documentation would imply that if you add the CORS filter without an `enabled` or `filter_enabled` key that the filter would not enforce CORS policies. However, this would not be accurate given [the default value](https://github.com/envoyproxy/envoy/blob/02ed57d648d82c5e5ef72070b145d2a28c9b404c/source/common/router/config_impl.cc#L140) of the `enabled` key.

Reply:
In istio, we set `filter_enabled` to `{}`, which is certainly possible in proto3 since all fields are optional. It sounds like we should revisit it and set `default_value` explicitly since that would avoid all confusion regarding the defaults.

Reply:
Yeah it might be worth being explicit for the time being. However, if you are setting `filter_enabled` to `{}` then the default value no longer applies since you're setting it to an empty dict.

I'm definitely open to revisiting this topic. I think it'll be less of a problem once the CORS config is migrated to use `per_route_config` since we can drop support for the `enabled` flag and remove any default behavior.